src/main/java/com/direwolf20/buildinggadgets/common/building/implementation/Column.java

```diff
@@ -0,0 +1,82 @@
+package com.direwolf20.buildinggadgets.common.building.implementation;
+
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.direwolf20.buildinggadgets.common.building.placement.IPlacementSequence;
+import com.direwolf20.buildinggadgets.common.tools.VectorTools;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumFacing.Axis;
+import net.minecraft.util.math.BlockPos;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.Spliterator;
+
+public class Column implements IPlacementSequence {
```

Comment:
Notice, that all non-final classes cannot be considered immutable, as any subclassing may corrupt this.